### PR TITLE
[core] Simplify placement code path for still and tile modes

### PR DESCRIPTION
**Pull request by pozdnyakov**
This PR removes from the `still` and `tile` map modes code path a bunch of logic dedicated to the symbol animation while placement.

### CODE REVIEW

File: src/mbgl/renderer/render_orchestrator.cpp [hunk 1]
```diff
@@ -377,12 +377,27 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(const UpdatePar
...
-            crossTileSymbolIndex.reset();
+    if (!isMapModeContinuous) {
+        crossTileSymbolIndex.reset();
+        renderTreeParameters->placementChanged = symbolBucketsChanged = !layersNeedPlacement.empty();
+        if (renderTreeParameters->placementChanged) {
+            Mutable<Placement> placement = makeMutable<Placement>(updateParameters.transformState,
+                                                                  updateParameters.mode,
+                                                                  updateParameters.transitionOptions,
+                                                                  updateParameters.crossSourceCollisions);
+            for (auto it = layersNeedPlacement.rbegin(); it != layersNeedPlacement.rend(); ++it) {
```
Comment (alexshalamov) [inline src/mbgl/renderer/render_orchestrator.cpp]: nit: crbegin / crend?

File: src/mbgl/renderer/render_orchestrator.cpp [hunk 2]
```diff
@@ -377,12 +377,27 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(const UpdatePar
     }
     // Symbol placement.
     bool symbolBucketsChanged = false;
-    {
-        if (!isMapModeContinuous) {
-            // TODO: Think about right way for symbol index to handle still rendering
-            crossTileSymbolIndex.reset();
+    if (!isMapModeContinuous) {
```
Comment (alexshalamov) [inline src/mbgl/renderer/render_orchestrator.cpp]: `if (!isMapModeContinuous) {` this is the most probable path, maybe move tile and static mode handling to `else` branch.

File: src/mbgl/text/placement.cpp [hunk 3]
```diff
@@ -534,11 +534,20 @@ void Placement::placeBucket(
 
 void Placement::commit(TimePoint now, const double zoom) {
     commitTime = now;
+    if (!getPrevPlacement()) {
```
Comment (alexshalamov) [inline src/mbgl/text/placement.cpp]: Is this only applicable for still & tile mode? Looks like in other cases, there is always valid previous placement provided by the controller. Maybe worth adding comment.

Comment (pozdnyakov) [reply src/mbgl/text/placement.cpp]: will add an assertion

**Pull request merged**